HousingFinanceInterimApi/V1/UseCase/GenerateReportUseCase.cs

```diff
@@ -0,0 +1,289 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using HousingFinanceInterimApi.V1.Domain;
+using HousingFinanceInterimApi.V1.Factories;
+using HousingFinanceInterimApi.V1.Gateways.Interface;
+using HousingFinanceInterimApi.V1.UseCase.Interfaces;
+using System.Threading.Tasks;
+using Google.Apis.Drive.v3.Data;
+using HousingFinanceInterimApi.V1.Boundary.Response;
+using HousingFinanceInterimApi.V1.Handlers;
+
+namespace HousingFinanceInterimApi.V1.UseCase
+{
+    public class GenerateReportUseCase : IGenerateReportUseCase
+    {
+        private readonly IBatchReportGateway _batchReportGateway;
+        private readonly IReportGateway _reportGateway;
+        //private readonly IReportAccountBalanceGateway _reportAccountBalanceGateway;
+        //private readonly IReportChargesGateway _reportChargesGateway;
+        //private readonly IReportCashImportGateway _reportCashImportGateway;
+        //private readonly IReportSuspenseAccountGateway _reportSuspenseAccountGateway;
+        private readonly IGoogleFileSettingGateway _googleFileSettingGateway;
+        private readonly IGoogleClientService _googleClientService;
+
+        private readonly string _waitDuration = Environment.GetEnvironmentVariable("WAIT_DURATION");
+        private readonly int _sleepDuration = 30000;
+
+        private const string ReportAccountBalanceByDateLabel = "ReportAccountBalanceByDate";
+        private const string ReportChargesLabel = "ReportCharges";
+        private const string ReportCashSuspenseLabel = "ReportCashSuspense";
+        private const string ReportCashImportLabel = "ReportCashImport";
+        private const string ReportHousingBenefitAcademyLabel = "ReportHousingBenefitAcademy";
+
+        public GenerateReportUseCase(IBatchReportGateway batchReportGateway,
+            IReportGateway reportGateway,
+            IGoogleFileSettingGateway googleFileSettingGateway,
+            IGoogleClientService googleClientService)
+        {
+            _batchReportGateway = batchReportGateway;
+            _reportGateway = reportGateway;
+            _googleFileSettingGateway = googleFileSettingGateway;
+            _googleClientService = googleClientService;
+        }
+
+        public async Task<StepResponse> ExecuteAsync()
+        {
+            LoggingHandler.LogInfo($"Checking if exist report in the queue");
+            var batchReports = await _batchReportGateway.ListPendingAsync().ConfigureAwait(false);
+
+            if (!batchReports.Any())
+                return new StepResponse() { Continue = false, NextStepTime = DateTime.Now.AddSeconds(int.Parse(_waitDuration)) };
+
+            var batchReport = batchReports.OrderBy(x => x.StartTime).First();
```

Comment:
What happens if batchReports is null?

Reply:
It means there's no report to process then the lambda will return false for the next step to finish the Step Function processing.